index.js

```diff
@@ -3,53 +3,225 @@ const state = {
+  const p = document.createElement('p')
+  const removeButton = document.createElement('button')
+  const span = document.createElement('span')
+  const addButton = document.createElement('button')
+  
+  img.setAttribute('class', 'cart--item-icon')
+  img.setAttribute('src', `assets/icons/${item.id}.svg`)
+  img.setAttribute('alt', `${item.name}`)
+
+  p.innerText = item.name
+
+  removeButton.setAttribute('class', 'quantity-btn remove-btn center')
+  removeButton.innerText = '-'
+  removeButton.addEventListener('click', () => {
+    item.quantity--
+    render()
+    })
+  span.setAttribute('class', 'quantity-text center')
+  span.innerText = item.quantity
+  
+  addButton.setAttribute('class', 'quantity-btn add-btn center')
+  addButton.innerText = '+'
+  addButton.addEventListener('click', () => {
+    item.quantity++
+    render()
+    })
+  li.append(img, p, removeButton, span, addButton)
+  ul.append(li)
+}
+
+function createFilterButton() {
+  const store = document.querySelector('#store')
+  const button = document.createElement('button')
+
+  button.setAttribute('class', 'filter-button')
+  button.innerText = 'Filter by Fruit'
+
+  button.addEventListener('click', () => {
+    filterOnlyFruit()
+  })
+  store.append(button)
+}
+
+function createSortButton() {
+  const store = document.querySelector('#store')
+  const button = document.createElement('button')
+
+  button.setAttribute('class', 'sort-button')
+  button.innerText = 'Sort A-Z'
+
+  button.addEventListener('click', () => {
+    sortAToZ()
+  })
+  store.append(button)
+}
+
+function render() {
+  const storeList = document.querySelector('.store--item-list')
+  const cartList = document.querySelector('.cart--item-list')
+  const total = document.querySelector('.total-section > div > .total-number')
+  
+  total.innerText = '0.00'
+  storeList.innerHTML = ''
+  cartList.innerHTML = ''
+  if(!document.querySelector('.filter-button')) {
+    createFilterButton()
+  }
+  if(!document.querySelector('.sort-button')) {
+    createSortButton()
+  }
+  state.items.forEach(item => {
+
+    if (state.filteredByFruit) {
+      if (item.type != 'fruit') {
+        return
+      }
+    }
+    
+    createStoreListItem(item)
+  });
+  state.cart.forEach(item => {
+    console.log('quantity of', item.name, '=', item.quantity)
+    total.innerText = (parseFloat(total.innerText) + (item.price * item.quantity)).toFixed(2)
+    if(item.quantity <= 0) {
+      return
+    }
+    createCartListItem(item)
+  })
+  
+}
+
+function filterOnlyFruit() {
+  if(state.filteredByFruit === true) {
+    state.filteredByFruit = false
+  }
+  else if(state.filteredByFruit === false) {
+    state.filteredByFruit = true
+  }
+  console.log('filter by fruit = ', state.filteredByFruit)
+  render()
+}
+
+function sortAToZ() {
+  if(state.sortedAToZ === true) {
+    state.sortedAToZ = false
+    state.items.sort ((a, b) => {
+      if (a.id < b.id) return -1;
+      if (a.id > a.id) return 1;
+      return 0;
```

Comment:
nice, i spent like 30minutes trying to figure out how to do it with nested for loops, then just resorted to this haha